testing/test_multicall.py

```diff
@@ -197,3 +197,40 @@ def m2():
     with pytest.raises(exc):
         MC([m2, m1], {}).execute()
     assert l == ["m1 init", "m1 finish"]
+
+
+@hookimpl(hookwrapper=True)
+def m1(arg1, arg2, arg3):
+    yield
+
+
+@hookimpl
+def m2(arg1, arg2, arg3):
+    return arg1, arg2, arg3
+
+
+@hookimpl(hookwrapper=True)
+def w1(arg1, arg2, arg3):
+    yield
+
+
+@hookimpl(hookwrapper=True)
+def w2(arg1, arg2, arg3):
+    yield
+
+
+def inner_exec(methods):
+    return MC(methods, {'arg1': 1, 'arg2': 2, 'arg3': 3}).execute()
+
+@pytest.mark.benchmark
+def test_hookimpls_speed(benchmark):
+    benchmark(inner_exec, [m1, m2])
+
+
+@pytest.mark.benchmark
+def test_hookwrapper_speed(benchmark):
+    benchmark(inner_exec, [w1, w2])
```

Comment:
Yes, right. I don't know much about the pluggy internals.

About displaying results, and running pytest-benchmark in CI ... well, it's not good. CI will run each with lots of external bias, so timings aren't consistent. You could look at perf counters or profile data (my plugin only shows some profile data but not well integrated in the stats yet) or you could compare against a baseline (eg, you have a test that only does 10 function calls to establish a baseline to compare against - if there's a perf regression you will get slower than the baseline.

Just to restate, saving benchmark data somewhere and comparing against previous runs would be unreliable due to virtualization and contention in TravisCI. We can talk about instruction counters I guess (since those shouldn't care about preemption right?), what would be useful?

And testing on pypy further complicates things because of warmup (the plugin has such feature but you need to set it correctly).